Increase test coverage for rfxtrx integration

RobBie1221 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This PR increases test coverage for rfxtrx integration from 92% to 96%. Additionally, while making tests, I found out that when incorrect event codes are entered in the configuration, the integration fails to setup without debug message. I made a small adjustment to let test pass, which now sets the integration up without that event code and debug information will show what code was failed to add.

Further improvements can possibly be made, but I do not want to create merge conflicts with the option/config flow PR.

FROM:
```
----------- coverage: platform linux, python 3.8.3-final-0 -----------
Name                                               Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------------
homeassistant/components/rfxtrx/__init__.py          222     32    86%   93-94, 99, 188-206, 213, 253, 315-316, 320, 324, 335, 339-340, 352-353, 375-395
homeassistant/components/rfxtrx/binary_sensor.py     119      5    96%   85, 90-91, 99, 215
homeassistant/components/rfxtrx/config_flow.py        13      0   100%
homeassistant/components/rfxtrx/const.py               6      0   100%
homeassistant/components/rfxtrx/cover.py              73      3    96%   39-40, 48
homeassistant/components/rfxtrx/light.py              88      4    95%   51-52, 60, 78
homeassistant/components/rfxtrx/sensor.py             98      8    92%   32, 39, 73-74, 76, 84, 96, 180
homeassistant/components/rfxtrx/switch.py             71      3    96%   50-51, 59
--------------------------------------------------------------------------------
TOTAL                                                690     55    92%
```
TO:
```
----------- coverage: platform linux, python 3.8.3-final-0 -----------
Name                                               Stmts   Miss  Cover   Missing
--------------------------------------------------------------------------------
homeassistant/components/rfxtrx/__init__.py          227     21    91%   93-94, 99, 191-209, 216, 258, 320-321, 329, 340, 344-345, 357-358
homeassistant/components/rfxtrx/binary_sensor.py     119      2    98%   85, 215
homeassistant/components/rfxtrx/config_flow.py        13      0   100%
homeassistant/components/rfxtrx/const.py               6      0   100%
homeassistant/components/rfxtrx/cover.py              73      1    99%   48
homeassistant/components/rfxtrx/light.py              88      2    98%   60, 78
homeassistant/components/rfxtrx/sensor.py             98      3    97%   74, 94, 178
homeassistant/components/rfxtrx/switch.py             71      0   100%
--------------------------------------------------------------------------------
TOTAL                                                695     29    96%
```

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] The code has been formatted using Black (`black --fast homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [X] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [X] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [X] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-asc+-review%3Aapproved

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

RobBie1221 committed:
Increase switch coverage

RobBie1221 committed:
Increase binary sensor coverage

RobBie1221 committed:
Final improvements

RobBie1221 committed:
Remove debug statement

ctalkington commented on tests/components/rfxtrx/test_sensor.py:
```diff
@@ -343,3 +344,11 @@ async def test_rssi_sensor(hass, rfxtrx):
     assert state.state == "-72"
+
+
+async def test_empty_conversions(hass, rfxtrx):
```
i would prefer this test not direct call fhe sensor file but rather created a situation/mock that cause code to be run and result in state being some value

ctalkington commented on homeassistant/components/rfxtrx/sensor.py:
```diff
@@ -26,14 +26,14 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-def _battery_convert(value):
```
this change shouldnt be needed if the one test commented on below is updated

elupus replied:
I agree. on that. Even if it's not changed. I think they should remain internally marked. pylint errors in unit tests for usage of internal functions can imho be ignored.

RobBie1221 committed:
Adjust test duplicate cover

RobBie1221 committed:
Remove None return test

RobBie1221 commented:
I dropped the test and changed it back to internal. At this moment, the gain of testing these two lines do not outweigh the effort needed to check how to mock something. We also first need to understand if and how rssi could be None without an exception being thrown somewhere. If it cannot happen, we are probably better off removing the None check code.

ctalkington commented on tests/components/rfxtrx/test_switch.py:
```diff
@@ -151,3 +152,22 @@ async def test_discover_rfy_sun_switch(hass, rfxtrx_automatic):
+    await hass.async_block_till_done()
+
+    conf_entries = hass.config_entries.async_entries(DOMAIN)
+    entry = conf_entries[0]
```
```suggestion
    assert len(conf_entries) == 1

    entry = conf_entries[0]
```

RobBie1221 committed:
Assert on length of conf_entries

Co-authored-by: Chris Talkington <chris@talkingtontech.com>
